pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+
+import { CardBody } from "@patternfly/react-core/dist/esm/components/Card/index.js";
+import { DescriptionList } from "@patternfly/react-core/dist/esm/components/DescriptionList/index.js";
+
+import { StorageCard, StorageDescription, new_card, new_page } from "../pages.jsx";
+import { StorageUsageBar } from "../storage-controls.jsx";
+import { get_fstab_config_with_client } from "../utils.js";
+import { btrfs_usage } from "./utils.jsx";
+import { mounting_dialog } from "../filesystem/mounting-dialog.jsx";
+import { check_mismounted_fsys, MismountAlert } from "../filesystem/mismounting.jsx";
+import { is_mounted, mount_point_text, MountPoint } from "../filesystem/utils.jsx";
+import client from "../client.js";
+
+const _ = cockpit.gettext;
+
+function subvolume_unmount(volume, subvol, forced_options) {
+    const block = client.blocks[volume.path];
+    mounting_dialog(client, block, "unmount", forced_options, subvol);
+}
+
+function subvolume_mount(volume, subvol, forced_options) {
+    const block = client.blocks[volume.path];
+    mounting_dialog(client, block, "mount", forced_options, subvol);
+}
+
+export function make_btrfs_subvolume_page(parent, volume, subvol) {
+    const actions = [];
+
+    const use = btrfs_usage(client, volume);
+    const block = client.blocks[volume.path];
+    const fstab_config = get_fstab_config_with_client(client, block, false, subvol);
+    const [, mount_point] = fstab_config;
+    const mismount_warning = check_mismounted_fsys(block, block, fstab_config, subvol);
+    const mounted = is_mounted(client, block, subvol);
+    const mp_text = mount_point_text(mount_point, mounted);
+    if (mp_text == null)
+        return null;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20240112-110325-d6d2c3a3-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)